src/visualizer/gui/panels/scene_panel.cpp

```diff
@@ -266,18 +266,23 @@ namespace lfs::vis::gui {
 
         // Context menu for folder
         theme().pushContextMenuStyle();
+
+        const bool blnTraininMode = m_trainerManager->hasTrainer();
+
         if (ImGui::BeginPopupContextItem("##ModelsMenu")) {
-            if (ImGui::MenuItem("Add PLY...")) {
-                cmd::ShowWindow{.window_name = "file_browser", .show = true}.emit();
+            if (!blnTraininMode) {
```

Comment:
You could just use here  m_trainerManager->hasTrainer();
so this never get queried         const bool blnTraininMode = m_trainerManager->hasTrainer();
when  if (ImGui::BeginPopupContextItem("##ModelsMenu")) { should be false